src/services/TechRecordsService.ts

```diff
@@ -335,6 +351,36 @@ class TechRecordsService {
     }
   }
 
+  public async archiveTechRecordStatus(systemNumber: string, techRecordToUpdate: ITechRecordWrapper, userDetails: IMsUserDetails) {
+    const allTechRecordWrapper: ITechRecordWrapper[] = await this.getTechRecordsList(systemNumber, STATUS.ALL, SEARCHCRITERIA.SYSTEM_NUMBER);
+    if(allTechRecordWrapper.length !== 1) {
+      // systemNumber search should return a single record
+      throw new HTTPError(400, ERRORS.NO_UNIQUE_RECORD);
+    }
+    const techRecordWithAllStatues = allTechRecordWrapper[0];
+    const filteredTechRecordToUpdate = techRecordWithAllStatues.techRecord.filter((techRecord) => techRecord.statusCode === techRecordToUpdate.techRecord[0].statusCode);
+    if (filteredTechRecordToUpdate.length === 1) {
+      if(filteredTechRecordToUpdate[0].statusCode === STATUS.ARCHIVED) {
+        throw new HTTPError(400, ERRORS.CANNOT_UPDATE_ARCHIVED_RECORD);
+      }
+      filteredTechRecordToUpdate[0].statusCode = STATUS.ARCHIVED;
+      filteredTechRecordToUpdate[0].lastUpdatedAt = new Date().toISOString();
+      filteredTechRecordToUpdate[0].lastUpdatedByName = userDetails.msUser;
+      filteredTechRecordToUpdate[0].lastUpdatedById = userDetails.msOid;
+      filteredTechRecordToUpdate[0].updateType = UPDATE_TYPE.TECH_RECORD_UPDATE;
+
+      let updatedTechRecord;
+      try {
+        updatedTechRecord = await this.techRecordsDAO.updateSingle(techRecordWithAllStatues);
+      } catch (error) {
+        throw new HTTPError(500, HTTPRESPONSE.INTERNAL_SERVER_ERROR);
```

Comment:
I think it would be better to follow the existing approach and throw the actual statusCode and error message